operatorcourier/validate.py

```diff
@@ -128,16 +128,33 @@ def _csv_spec_validation(self, spec, bundleData):
 
             if "owned" not in customresourcedefinitions:
                 logger.error("spec.customresourcedefinitions.owned not defined for csv")
-                valid = False
-            else:
-                for crd in customresourcedefinitions["owned"]:
-                    if "name" not in crd:
-                        logger.error("name not defined for item in spec.customresourcedefinitions.")
-                        valid = False
-                    else:
-                        if crd["name"] not in crdList:
-                            logger.error("custom resource definition referenced in csv not defined in root list of crds")
-
+                return False
+
+            for csvOwnedCrd in customresourcedefinitions["owned"]:
+                if "name" not in csvOwnedCrd:
```

Comment:
Should we create a list like ["name", "kind", ..] and merge these if statements into a forloop with an if statement?

Thoughts @kevinrizza @jeremylinlin

Reply:
I created this [helper](https://github.com/operator-framework/operator-courier/pull/41/files/a9b0bbdf052014dd5ce4a696e3d1f15f69cdf8a3#diff-4badd63917dd32ab713b1c89ddfa260cR52) that recursively checks the presence of [required nested fields](https://github.com/operator-framework/operator-courier/pull/41/files#diff-4badd63917dd32ab713b1c89ddfa260cR11) in yaml files. It should work for both CRDs and CSVs. Maybe I should do it in a separate PR for [this issue](https://github.com/operator-framework/operator-courier/issues/17)? @awgreene